src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11085,133 +11085,14 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax unaryOrPrimar
                 }
 
                 if (IsExpectedAssignmentOperator(operatorToken.Kind))
-                {
-                    ExpressionSyntax rhs;
-
-                    if (operatorExpressionKind == SyntaxKind.SimpleAssignmentExpression && CurrentToken.Kind == SyntaxKind.RefKeyword &&
-                        // check for lambda expression with explicit ref return type: `ref int () => { ... }`
-                        !this.IsPossibleLambdaExpression(newPrecedence))
-                    {
-                        rhs = _syntaxFactory.RefExpression(
-                            this.EatToken(),
-                            this.ParseExpressionCore());
-                    }
-                    else
-                    {
-                        rhs = this.ParseSubExpression(newPrecedence);
-                    }
-
-                    return _syntaxFactory.AssignmentExpression(
-                        operatorExpressionKind, leftOperand, operatorToken, rhs);
-                }
+                    return ParseAssignmentOperator(operatorExpressionKind, leftOperand, operatorToken);
```

Comment:
I extracted shared logic from `ParseExpressionContinued`, including
- body of this `if`
- local function `getOperatorTokenAndExpressionKind`
- local function `eatOperatorToken`

Reply:
Please see https://github.com/dotnet/roslyn/commit/b68ecbaf2e1a1256e35d89d6792bc1cf91e16249?w=1 for diffs of the extracted sections.

Reply:
note: your life may be easier if you have an earlier pr which does this extraction anyways.  but up to you.